### PR TITLE
Feature/reusable pagination on SavedData

### LINKED ISSUE #10: Create saved-boards page

### PR DESCRIPTION
**Pull request by tomivm**
close #10

### CODE REVIEW

**Review by RodriSanchez1 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/dashboard/@savedBoards/SavedBoardList.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    (histories, historyToDelete: SavedBoardsData) => {
+      return histories.filter((h) => h.id != historyToDelete.id);
+    },
+  );
+  const deleteSavedBoardsData = async (historyToDelete: SavedBoardsData) => {
+    deleteHistory(historyToDelete);
+    await removeSavedBoardsData(historyToDelete);
+  };
+
+  return <DataList list={histories} deleteItem={deleteSavedBoardsData} />;
```
Comment (maxicapodacqua) [inline src/app/[locale]/dashboard/@savedBoards/SavedBoardList.tsx:24]: Rename all `histories` variables to a more meaningful name (savedBoards?)

File: src/components/DataList/DataList.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  stack: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+    width: '100%',
+  },
+  list: { width: '100%' },
+  pagination: { paddingBottom: 1 },
+};
```
Comment (maxicapodacqua) [inline src/components/DataList/DataList.tsx:20]: Consider adding `satisfies Record<string, SxProps>` to insure types

File: src/components/DataList/DataList.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  deleteItem,
+}: {
+  list: DataItemType[];
+  deleteItem: (data: DataItemType) => void;
+}) {
+  const [page, setPage] = useState(1);
+  const handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
+    setPage(value);
+  };
+  const itemsPerPage = 2;
```
Comment (maxicapodacqua) [inline src/components/DataList/DataList.tsx:33]: Consider creating a global constant for this, I used upper-cased constants in a separate file (like constants.ts) in my current branch

**Review by maxicapodacqua — CHANGES_REQUESTED** [review]

File: src/app/[locale]/dashboard/@savedBoards/actions.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    date: yesterday.toISOString(),
+  },
+];
+
+export async function getSavedBoardsData(): Promise<SavedBoardsData[]> {
+  // Fetching with a 2 seconds delay to test loading state
+  await fetch('https://postman-echo.com/delay/2', {
+    cache: 'no-cache',
+  });
+  // Here we query DB
```
[the remark this answers is not in the document]
Comment (tomivm) [reply src/app/[locale]/dashboard/@savedBoards/actions.ts:58]: It was resolved Using the error.js file convention. Please show me how to add the translations to it. @RodriSanchez1

File: src/components/ErrorAndReset/Styles.ts
Comment (RodriSanchez1) [inline src/components/ErrorAndReset/Styles.ts:1]: Change file name to `styles.ts`

File: src/components/ErrorAndReset/ErrorAndReset.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+          }
+          sx={Styles.iconButton}
+        >
+          <RefreshIcon />
+        </IconButton>
+      }
+      sx={{ m: 2 }}
+    >
+      <AlertTitle>ERROR</AlertTitle>
+      Something went wrong getting your data!
```
Comment (RodriSanchez1) [inline src/components/ErrorAndReset/ErrorAndReset.tsx:26]: Use intl library for the messages.

File: src/components/DataList/Styles.ts
Comment (RodriSanchez1) [inline src/components/DataList/Styles.ts:1]: Change name to `styles.ts`

File: src/components/ErrorAndReset/ErrorAndReset.tsx (hunk 5 above)
Comment (tomivm) [reply src/components/ErrorAndReset/ErrorAndReset.tsx:26]: Please abord this in a different PR 
Issue: https://github.com/cboard-org/cboard-ai-builder/issues/54

**Pull request merged**